staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -86,6 +86,8 @@ type Config struct {
 	// Server should be accessed without verifying the TLS
 	// certificate. For testing only.
 	Insecure bool
+	// ServerName is an optional override passed to the server for SNI and used to check server ceritificates against.
+	ServerName string
```

Comment:
having both Host and ServerName is a little crazy... can this go in TLSClientConfig or be named TLSServerName?